modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/webpack_federation.js

```diff
@@ -12,61 +12,198 @@
  * details.
  */
 
-Liferay = window.Liferay || {};
+(function() {
+	const CONTAINER_REQUESTS_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_CONTAINER_REQUESTS__');
+	const CONTAINERS_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_CONTAINERS__');
+	const GET_MODULE_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_GET_MODULE__');
+	const SHARED_SCOPE_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_SHARED_SCOPE__');
 
-Liferay.Webpack = {
-	Container: {},
-	SharedScope: {},
-};
+	window[CONTAINER_REQUESTS_SYMBOL] = window[CONTAINER_REQUESTS_SYMBOL] || {};
+	window[CONTAINERS_SYMBOL] = window[CONTAINERS_SYMBOL] || {};
+	window[SHARED_SCOPE_SYMBOL] = window[SHARED_SCOPE_SYMBOL] || {};
 
-Liferay.require = (moduleName) =>
-	new Promise((resolve, reject) => {
-		const i = moduleName.indexOf('/');
+	const containerRequests = window[CONTAINER_REQUESTS_SYMBOL];
+	const sharedScope = window[SHARED_SCOPE_SYMBOL];
 
-		let containerId, path;
+	/**
+	 * Create a new container request
+	 */
+	function createContainerRequest(url, onLoadHandler) {
+		const script = document.createElement('script');
 
-		if (i === -1) {
-			containerId = moduleName;
-			path = '.';
-		}
-		else {
-			containerId = moduleName.substring(0, i);
-			path = moduleName.substring(i + 1);
-		}
+		const containerRequest = {
+			// set if request failed
+			error: undefined,
+			// set to true if the .js file has been fetched from the server
+			fetched: false,
+			// set to the exported value (if error is set module is undefined)
+			module: undefined,
+			// set to the <script> DOM node while it is being retrieved (undefined after)
+			script,
+			// set to all people that needs to be notified once the <script> loads
+			subscribers: [],
+		};
 
-		let container = Liferay.Webpack.Container[containerId];
+		script.src = url;
+		script.onload = () => onLoadHandler(containerRequest);
 
-		if (container) {
-			Promise.resolve(container.get(path)).then((module) =>
-				resolve(module())
-			);
-		}
-		else {
-			const script = document.createElement('script');
+		document.body.appendChild(script);
+
+		return containerRequest;
+	}
+
+	/**
+	 * Log messages if `explain resolutions` option is turned on.
+	 */
+	function explain(...things) {
+		// Until we get to the end of webpack 5 migration we will log all
+		// messages no matter what is configured. Once finished, we will only
+		// write when the old existing `explain resolutions` option is on.
+		console.log(...things);
```

Comment:
Linter will complain here unless you add a suppression I think.

Reply:
I'll change this because it is filling the console with a lot of messages. I logged very few things when this began, but now it is dumping a big manifest.